pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxReceiveOperator.java

```diff
@@ -176,6 +214,18 @@ protected TransferableBlock getNextBlock() {
       }
     }
 
+    if (((openMailboxCount == 0) || (openMailboxCount <= eosMailboxCount))
+        && (!CollectionUtils.isEmpty(_priorityQueue)) && !_isSortedBlockConstructed) {
+      // Some data is present in the PriorityQueue, these need to be sent upstream
+      LinkedList<Object[]> rows = new LinkedList<>();
+      while (_priorityQueue.size() > 0) {
+        Object[] row = _priorityQueue.poll();
+        rows.addFirst(row);
+      }
+      _isSortedBlockConstructed = true;
+      return new TransferableBlock(rows, _dataSchema, DataBlock.Type.ROW);
+    }
```

Comment:
based on this logic. as long as there's a _priorityQueue and size is not zero you should return the priority queue results. 
when any mailbox receives an error. it should also dropped any _priorityQueue content and stopped the consumption.

Reply:
Agreed, I've modified the code to drop the contents of the `_priorityQueue` on receiving an error block. Keeping this block as is. Let me know if you intended for a different change here.